### PR TITLE
Added table component, not finished

### PR DESCRIPTION
**Pull request by MartinezBaker**
Reverted back to using table elements.. the problem with the expanding progress bar was more than likely coming from the progress bar component it's self. I switched the sizing of the parent div from width: "100%" to width: "190px". I believed this helped because from what I could see the sizing of the parent div was dependent on the size of the table cell, and the size of the table cell was dependent on its contents, giving the progress bar no real boundaries, so by setting it to a fixed width it seemed to fix the problem. Once again sizing and spacing isn't set in stone, these are just 'raw renders'.

### CODE REVIEW

**Comment by MartinezBaker** [comment]
Made requested changes

File: crypto-app/src/components/CoinInstance/CoinInstance.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCaretDown, faCaretUp } from "@fortawesome/free-solid-svg-icons";
+import { TableCharts, ProgressBar } from "components";
+import {formatTimePercent, formatNum} from '../../utils/index';
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/components/CoinInstance/CoinInstance.js]: import {` `formatTimePercent, formatNum` `} from '../../utils/index';

File: crypto-app/src/components/CoinInstance/CoinInstance.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import { CoinImage, PercentColor, DataRow, TableCell } from './styles';
+
+const CoinInstance = (props) => (
+  <DataRow>
+    <TableCell>
+      <CoinImage src={props.image} alt="Coin" />{" "}
+      {props.name.length < 9 ? props.name : props.name.slice(0, 9) + "..."} (
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/components/CoinInstance/CoinInstance.js]: you could make an utility function here

File: crypto-app/src/components/CoinInstance/styles.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  padding: 15px;
+`;
+
+export const CoinImage = styled.img`
+    width: 40px;
+    vertical-align: middle;
+    margin-right: 5px;
+`
+export const PercentColor = styled.td`
+  color: ${(props) => {
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/components/CoinInstance/styles.js]: you can destructure here ({ data })

File: crypto-app/src/pages/Coins/Coins.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { CoinInstance } from "components";
+import InfiniteScroll from 'react-infinite-scroll-component';
+import { TableContainer, TableHeader, Table, TableRow } from './styles';
+import axios from 'axios';
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: npm modules come before your components

File: crypto-app/src/pages/Coins/Coins.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    }
+  }
+  getMoreCoins = async () => {
+    try {
+      this.setState({
+        page: this.state.page + 1,
+        isLoading: true,
+      });
+      const  { data }  = await axios(
+        `https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=${this.state.page}&sparkline=true&price_change_percentage=1h%2C24h%2C7d`
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js:47]: this.state.page in your query will be the previous page, not the one you setup in the beginning FYI

File: crypto-app/src/pages/Coins/Coins.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+      this.getCoins()
+  }
+  render(){
+    return (
+      <>
+        <TableContainer>
+          <InfiniteScroll
+            dataLength={this.state.coins.length}
+            next={this.getMoreCoins}
+            hasMore={true}
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: what if you get to the bottom?

File: crypto-app/src/pages/Coins/Coins.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+                        ? coin.current_price.toString()
+                        : coin.current_price.toLocaleString()
+                    }
+                    image={coin.image}
+                    symbol={coin.symbol.toUpperCase()}
+                    oneHour={
+                      coin.price_change_percentage_1h_in_currency === null
+                        ? null
+                        : coin.price_change_percentage_1h_in_currency.toString()
+                    }
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/Coins/Coins.js]: `oneHour={coin.price_change_percentage_1h_in_currency?.toString()}`

this might help

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: crypto-app/src/components/index.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export {default as CoinInstance} from './CoinInstance'
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/components/index.js]: export {` `default as CoinInstance` `} from './CoinInstance'

File: crypto-app/src/pages/CoinPage/index.js [hunk 9]
```diff
@@ -0,0 +1 @@
+export {default} from './CoinPage'
```
Comment (yoloOnTheBattlefield) [inline crypto-app/src/pages/CoinPage/index.js]: export {` `default` `} from './CoinPage'

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

**Pull request merged**